src/components/helpers/dv_panel_helper.ts

```diff
@@ -0,0 +1,327 @@
+import { ComponentConstructor } from "@odoo/owl";
+import { Action, ActionSpec, createActions } from "../../actions/action";
+import { getFormattedDate } from "../../helpers";
+import { Registry } from "../../registries/registry";
+import { _t } from "../../translation";
+import {
+  DataValidationCriterion,
+  DataValidationCriterionType,
+  DateIsAfterCriterion,
+  DateIsBeforeCriterionCriterion,
+  DateIsBetweenCriterion,
+  DateIsCriterionCriterion,
+  DateIsNotBetweenCriterion,
+  DateIsOnOrAfterCriterion,
+  DateIsOnOrBeforeCriterion,
+  SpreadsheetChildEnv,
+} from "../../types";
+import { DataValidationDateCriterionForm } from "../side_panel/data_validation/dv_criterion_form/dv_date_criterion/dv_date_criterion";
+import { DataValidationDoubleInputCriterionForm } from "../side_panel/data_validation/dv_criterion_form/dv_double_input_criterion/dv_double_input_criterion";
+import { DataValidationNoInput } from "../side_panel/data_validation/dv_criterion_form/dv_no_input_criterion/dv_no_input_criterion";
+import { DataValidationSingleInputCriterionForm } from "../side_panel/data_validation/dv_criterion_form/dv_single_input_criterion/dv_single_input_criterion";
+import { DataValidationListCriterionForm } from "../side_panel/data_validation/dv_criterion_form/dv_value_in_list_criterion/dv_value_in_list_criterion";
+import { DataValidationValueInRangeCriterionForm } from "../side_panel/data_validation/dv_criterion_form/dv_value_in_range_criterion/dv_value_in_range_criterion";
+import { DVTerms } from "../translations_terms";
+
+export type DVCriterionCategory = "text" | "date" | "number" | "misc";
+export const dvCriterionCategoriesSequences: Record<DVCriterionCategory, number> = {
+  text: 20,
+  date: 30,
+  number: 40,
+  misc: 50,
+};
+
+export type DataValidationCriterionItem = {
+  type: DataValidationCriterionType;
+  component: ComponentConstructor;
+  name: string;
+  sequence: number;
+  getPreview: (criterion: DataValidationCriterion, env: SpreadsheetChildEnv) => string;
+  category: DVCriterionCategory;
+};
+
+export const dataValidationPanelCriteriaRegistry: Registry<DataValidationCriterionItem> =
+  new Registry();
+
+dataValidationPanelCriteriaRegistry.add("textContains", {
```

Comment:
I would personnlay find it more easier to have each component registering itself in the registry, so that when we add one new rule (and the related component) we wont forget to add it here.